service-statistics-sotka/src/main/java/fi/nls/oskari/control/statistics/plugins/sotka/SotkaIndicatorValuesFetcher.java

```diff
@@ -6,105 +6,131 @@
+                    break;
+                default:
             }
         }
         try {
             request.setGender(gender);
             String[] yearsArray = years.toArray(new String[years.size()]);
             request.setYears(yearsArray);
             request.setIndicator(indicator);
-            return parser.parse(request.getData());
-            
+            return parseJSON(request.getData());
+
         } catch (JSONException e) {
             e.printStackTrace();
             throw new APIException("Something went wrong parsing JSON from SotkaNET getIndicatorValues API.", e);
         }
     }
 
     /**
-     * @param selectors Used to query SotkaNET with.
-     * @param indicator The indicator we want.
-     * @param regionCategoryId The oskari layer we are interested in. For example: "KUNTA"
+     * The response is indexed by region id.
+     *
+     * @param response
      * @return
+     * @throws JSONException
      */
-    public Map<String, IndicatorValue> get(StatisticalIndicatorDataModel selectors, String indicator,
-                                           String regionCategoryId) {
-        Map<Integer, IndicatorValue> allValues = getAll(selectors, indicator);
-        Map<String, IndicatorValue> filteredValues = new HashMap<>();
-        for (Entry<Integer, IndicatorValue> entry: allValues.entrySet()) {
-            Integer regionId = entry.getKey();
-            if(regionId == null) {
-                // sometimes this is null with 0.0 as value.
-                // TODO: check if this is a bug in some other code or an empty row from sotkanet
-                continue;
-            }
-            
-            IndicatorValue value = entry.getValue();
-            // SotkaNET gives "Kunta" in some places, "KUNTA" in others...
-            String category = regionParser.getCategoryById(regionId);
-            if (category != null && category.equalsIgnoreCase(regionCategoryId)) {
-                // We must include this value to the result.
-                String code = regionParser.getCode(regionId);
-                if (code != null) {
-                    filteredValues.put(code, value);
-                }
+    private Map<Integer, IndicatorValue> parseJSON(String response) throws JSONException {
+        Map<Integer, IndicatorValue> indicatorMap = new HashMap<>();
+        // The response is a String JSON array with JSONObjects
+        JSONArray responseArray = new JSONArray(response);
+        for (int i = 0; i < responseArray.length(); i++) {
+            // Example row: {"indicator" : 4,"region": 231,"year": 2012,"gender": "total","value": 3.4,"absValue": 9}
+            JSONObject valueRow = responseArray.getJSONObject(i);
+            Double doubleValue = valueRow.optDouble("value");
+            if (Double.NaN != doubleValue) {
+                IndicatorValue indicatorValue = new IndicatorValueFloat(doubleValue);
+                Integer id = Integer.valueOf(valueRow.getString("region"));
```

Comment:
If the "region" is always an integer could just call `valueRow.getInt("region");`